src/settings/eeprom/mod.rs

```diff
@@ -0,0 +1,21 @@
+pub mod main_board;
+pub mod rf_channel;
+mod sinara;
+
+use encdec::{Decode, DecodeOwned, Encode};
+use serde::{Deserialize, Serialize};
+
+/// A semantic version control for recording software versions.
+#[derive(Encode, DecodeOwned, Serialize, Deserialize, Debug, PartialEq, Copy, Clone)]
+pub struct SemVersion {
+    major: u8,
+    minor: u8,
+    patch: u8,
+}
+
+impl SemVersion {
+    /// Determine if this version is compatible with `rhs`.
```

Comment:
It's not symmetric. The wording is wrong.

Reply:
I'm not sure what you're saying here. This is the semver comparison logic we've been using to check the old EEPROM settings for upgrade and it's been working fine (this is just a restructure of existing code). What would you prefer the docs to be?

Reply:
The wording "A compatible with B" implies "B compatible with A". The code doesn't match that.
LHS and RHS are different things here, not "this version". LHS is the requirement, RHS is the candidate. They can't be interchanged.

Reply:
"Determine if the candidate version `rhs` if compatible with the requirement `lhs`."

Reply:
Yeah this comparison is broken and the code compensates for this. I'm cleaning up the im

Reply:
Fixed in 088b10202ef98a6dbe41c527f801c147a0cd082a